src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -163,10 +163,11 @@ public class ControllerArgumentBinder : IControllerArgumentBinder
                 parameter.BindingInfo,
                 parameter.Name);
 
-            if (parameter.BindingInfo?.BinderModelName != null)
+            var parameterModelName = parameter.BindingInfo?.BinderModelName ?? metadata.BinderModelName;
```

Comment:
Found it when I added the `[ModelBinder(Name = "HomeAddress")]` tests in `MutableObjectModelBinderIntegrationTest`.